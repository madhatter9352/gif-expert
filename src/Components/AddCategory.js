import React, {useState} from 'react'
import PropTypes from 'prop-types';

const AddCategory = ({setCateg}) => {
    const [inputValue, setInputValue] = useState("")

    const handleInputValue = (e) => {
        setInputValue(e.target.value);
    }
    
    const handleSubmit = (e) => {
        e.preventDefault();

        if (inputValue.trim().length > 2){
            setCateg(cate => [inputValue, ...cate]);
            setInputValue("");
        }

    }

    return (
        <form onSubmit = { handleSubmit }>
            <div>
                <input 
                    type = "text"
                    value = { inputValue }
                    onChange = { handleInputValue }
                />
            </div>
        </form>
    )
}

AddCategory.propTypes = {
    setCateg: PropTypes.func.isRequired 
}

export default AddCategory
