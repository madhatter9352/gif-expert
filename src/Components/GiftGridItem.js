import React from 'react'
import PropTypes from 'prop-types';

export const GiftGridItem = ( {id, title, url} ) => {
    console.log(id, title, url);
    return (
        <div className="card animate__animated animate__bounce">
            <img src = {url} alt={title} width={250} style={{padding: 10}}></img>
            <p>{title}</p>
        </div>
    )
}

GiftGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}
