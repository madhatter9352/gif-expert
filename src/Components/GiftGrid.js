import React from 'react'
import {GiftGridItem} from './GiftGridItem'

import { useFectchGifs } from '../hooks/useFetchGifs';

export const GiftGrid = ( { category } ) => {
    
    // const [images, setImages] = useState([]);
    const {data:images, loading} = useFectchGifs(category);
    // useEffect(() => {
    //     getGifs(category)
    //         .then( imgs => setImages(imgs))
    // }, [category])

    return (
        <>
        <h3>{category}</h3>
        { loading && <h1>Loading</h1> }
        <div className="card-grid">
            {images.map( gif => (
                    <GiftGridItem 
                        key={gif.id}
                        {... gif}
                    />
                ))}
        </div>
        </>
    )
}
