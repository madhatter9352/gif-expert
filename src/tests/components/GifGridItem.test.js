import React from 'react';
import { shallow } from 'enzyme';
import { GiftGridItem } from '../../Components/GiftGridItem';

describe('Pruebas en GifGrdiItem', () => {
    const title = 'Un titulo';
    const url = 'https://localhost.com'
    const wraper = shallow(<GiftGridItem title={title} url={url}/>)

    test('debe mostrar el componente correctamente', () => {
        expect(wraper).toMatchSnapshot()
    })

    test('debe tener un parrafo con el titulo', () => {
        const p = wraper.find('p')
        expect(p.text().trim()).toBe(title)
    })
    
    test('debe tener la imagen igual al url y alt de los props', () => {
        const img = wraper.find('img');

        // console.log(img.prop('src'));

        expect(img.prop('src')).toBe(url)
        expect(img.prop('alt')).toBe(title)
    })

    test('debe de tener animate__bounce', () => {
        const div = wraper.find('div')
        const className = div.prop('className')
        expect(className.includes('animate__bounce')).toBe(true)
    })
    
})
