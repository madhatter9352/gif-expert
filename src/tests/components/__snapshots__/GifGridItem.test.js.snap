// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGrdiItem debe mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__bounce"
>
  <img
    alt="Un titulo"
    src="https://localhost.com"
    style={
      Object {
        "padding": 10,
      }
    }
    width={250}
  />
  <p>
    Un titulo
  </p>
</div>
`;
